plugins/es.upv.paella.playPauseButtonPlugin/playbutton_onscreen.js

```diff
@@ -60,9 +68,16 @@ paella.addPlugin(function() {
 		}
 	
 		endVideo() {
-			this.isPlaying = false;
-			this.showIcon = this.showOnEnd;
-			this.checkStatus();
+			paella.player.videoContainer.ended()
+			.then(ended => {
+				if (ended) {
+					this.isPlaying = false;
+					this.showIcon = this.showOnEnd;
+					this.checkStatus();
+				} else {
+					base.log.debug(`BTN ON SCREEN: The player is no longer in ended state.`);
+				}
+			});
```

Comment:
This sets the play button state correctly when the player ended event occurs, as well as when the endVideo event occurs.